akka-sample-kafka-to-sharding/processor/src/main/scala/sample/sharding/kafka/TopicListener.scala

```diff
@@ -0,0 +1,49 @@
+package sample.sharding.kafka
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.external._
+import akka.cluster.typed.Cluster
+import akka.kafka.ConsumerRebalanceEvent
+import akka.kafka.TopicPartitionsAssigned
+import akka.kafka.TopicPartitionsRevoked
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+
+import scala.concurrent.duration._
+import scala.util.Failure
+import scala.util.Success
+
+object TopicListener {
```

Comment:
Is this something that we would add to Alpakka Kafka once there's a version of Akka Cluster available with external sharding?

Reply:
I think so as we can add in properly error handling.
Should be in a separate module so that we don't pull in sharding for all users.

Reply:
Yes, we could make akka sharding a provided dep.  We do something similar for the Kafka service discovery feature.

https://doc.akka.io/docs/alpakka-kafka/current/discovery.html